hardhat.config.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @type import('hardhat/config').HardhatUserConfig
+ */
+ require('hardhat/config')
+ require('dotenv').config()
+ require("@nomiclabs/hardhat-etherscan")
+ require("@nomiclabs/hardhat-ethers")
+ require("hardhat-deploy-ethers")
+ require('@openzeppelin/hardhat-upgrades')
+ 
+ 
+ 
+ task(
+   "deploy",
+   "Deploy your MasterChef contract",
+   async (_, { ethers, upgrades }) => {
+   
+     const artifact = await ethers.getContractFactory("MasterChef");
+     const contract = await upgrades.deployProxy(artifact,["YOU CONSTRUCTOR ARGUMENT HERE"]);
+     
+     console.log('YOUR PROXY ADDRESS :', contract.address)
+     console.log('⚠️ Deployment of your main proxy successful. Please Copy the Proxy address to your hardhat-config.js file before continuing.')
+     
+     // after deployment you should pass the ownership to the Gnosis Vault
```

Comment:
Why the Gnosis Vault?

Reply:
Gnosis vault are multi signature wallet where you usually transfer ownership of big project contract to add a layer of security in case the admin key becomes compromise.  It's also compatible with OZ defender.